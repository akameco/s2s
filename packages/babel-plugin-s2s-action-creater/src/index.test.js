// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

// const filename = path.resolve(process.cwd(), 'app', 'counter', 'actionTypes.js')
const filename = path.resolve(__dirname, '__fixtures__', 'actionTypes.js')

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: { filename, configFile: false },
  tests: [
    {
      title: 'one function',
      code: `
// @flow
export const INCREMENT: "counter/INCREMENT" = "counter/INCREMENT";

export const Actions = {
  INCREMENT
};

export type Increment = {
  type: typeof INCREMENT
};

export type Action = Increment;
      `,
    },
    {
      title: 'some function',
      code: `
export const INCREMENT: 'App/INCREMENT' = 'App/INCREMENT'
export const DECREMENT: 'App/DECREMENT' = 'App/DECREMENT'

export const Actions = {
  INCREMENT,
  DECREMENT,
}

export type Increment = {
  type: typeof INCREMENT,
}

export type Decrement = {
  type: typeof DECREMENT,
}

export type Action = Increment | Decrement
      `,
    },
    {
      title: 'has prop',
      code: `
// @flow
export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  payload: number,
}

export type Action = Add
    `,
    },
    {
      title: 'another import',
      code: `
// @flow
import type { Count } from './types'

export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  n: number,
  payload: Count,
}

export type Action = Add
    `,
    },
  ],
})
