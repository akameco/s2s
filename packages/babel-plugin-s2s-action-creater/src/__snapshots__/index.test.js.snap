// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default another import: another import 1`] = `
"
// @flow
import type { Count } from './types'

export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  n: number,
  payload: Count,
}

export type Action = Add

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { Count } from './types';

import { ADD } from './actionTypes';
import type { Add } from './actionTypes';

export function add(n: number, payload: Count): Add {
  return {
    type: ADD,
    n,
    payload
  };
}
"
`;

exports[`default has prop: has prop 1`] = `
"
// @flow
export const ADD: 'App/ADD' = 'App/ADD'

export const Actions = {
  ADD,
}

export type Add = {
  type: typeof ADD,
  payload: number,
}

export type Action = Add

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { ADD } from './actionTypes';
import type { Add } from './actionTypes';

export function add(payload: number): Add {
  return {
    type: ADD,
    payload
  };
}
"
`;

exports[`default one function: one function 1`] = `
"
// @flow
export const INCREMENT: \\"counter/INCREMENT\\" = \\"counter/INCREMENT\\";

export const Actions = {
  INCREMENT
};

export type Increment = {
  type: typeof INCREMENT
};

export type Action = Increment;

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { INCREMENT } from \\"./actionTypes\\";
import type { Increment } from \\"./actionTypes\\";

export function increment(): Increment {
  return {
    type: INCREMENT
  };
}
"
`;

exports[`default some function: some function 1`] = `
"
export const INCREMENT: 'App/INCREMENT' = 'App/INCREMENT'
export const DECREMENT: 'App/DECREMENT' = 'App/DECREMENT'

export const Actions = {
  INCREMENT,
  DECREMENT,
}

export type Increment = {
  type: typeof INCREMENT,
}

export type Decrement = {
  type: typeof DECREMENT,
}

export type Action = Increment | Decrement

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { INCREMENT, DECREMENT } from './actionTypes';
import type { Increment, Decrement } from './actionTypes';

export function increment(): Increment {
  return {
    type: INCREMENT
  };
}
export function decrement(): Decrement {
  return {
    type: DECREMENT
  };
}
"
`;
